zerver/tests/test_custom_profile_data.py

```diff
@@ -601,17 +601,24 @@ def test_update_profile_data_successfully(self) -> None:
             ("GitHub", "zulip-mobile"),
         ]
 
-        data = []
+        data: List[ProfileDataElementUpdateDict] = []
+        expected_value: Dict[int, ProfileDataElementValue] = {}
+        expected_rendered_value: Dict[int, Optional[str]] = {}
         for i, field_value in enumerate(fields):
             name, value = field_value
             field = CustomProfileField.objects.get(name=name, realm=realm)
             data.append(
                 {
                     "id": field.id,
                     "value": value,
-                    "field": field,
                 }
             )
+            expected_value[field.id] = value
+            expected_rendered_value[field.id] = (
+                markdown_convert(value).rendered_content
+                if field.is_renderable() and isinstance(value, str)
```

Comment:
The `isinstance` check is necessary for typing because `.is_renderable` does not narrow `value`.